jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/MultiPart.java

```diff
@@ -57,10 +56,33 @@ public interface Parser
     {
         void deleteParts();
 
-        Part getPart(String name) throws IOException;
+        jakarta.servlet.http.Part getPart(String name) throws IOException;
 
-        Collection<Part> getParts() throws IOException;
+        Collection<jakarta.servlet.http.Part> getParts() throws IOException;
 
         List<ComplianceViolation.Event> getNonComplianceWarnings();
     }
+
+    /**
+     * Expose File of MultiPart section.
+     *
+     * @deprecated this interface is not available in ee10 or newer.
+     */
+    @Deprecated(since = "12.0.12", forRemoval = true)
+    public interface Part extends jakarta.servlet.http.Part
+    {
+        /**
+         * Get the file
+         *
+         * @return the file, if any, the data has been written to.
+         */
+        File getFile();
```

Comment:
Whilst I know this is already deprecated and not supported in ee10 and beyond.... but rather than a `getFile` semantics, would it not be better to have a `takeFile` semantic that was equivalent to calling `write(filename)` with the same filename.

Reply:
Note that `getFile()` is an existing API method on the individual implementations already.
This is just an interface that exposes it.

Reply:
I understand what it does and that it was an existing internal API.   I'm suggesting that if we are to make it public, then we should do so with a different method with take semantics.  We don't want users calling internal methods else that constrains how we can modify the code in future.